src/OrbitGl/GraphTrack.cpp

```diff
@@ -37,24 +37,27 @@ GraphTrack<Dimension>::GraphTrack(CaptureViewElement* parent,
     : Track(parent, timeline_info, viewport, layout, module_manager, capture_data),
       series_{series_names, series_value_decimal_digits, std::move(series_value_units)} {}
 
+template <size_t Dimension>
+bool GraphTrack<Dimension>::HasLegend() const {
+  return !IsCollapsed() && Dimension > 1;
+}
+
 template <size_t Dimension>
 float GraphTrack<Dimension>::GetHeight() const {
-  // Top content margin is counted twice because it it inserted before and after the legend
-  float height = layout_->GetTrackTabHeight() + layout_->GetTrackContentTopMargin() * 2 +
-                 GetLegendHeight() + GetGraphContentHeight() +
-                 layout_->GetTrackContentBottomMargin();
-  return height;
+  // Top content margin is counted twice when there are legends because it is inserted above and
+  // below the legend.
+  float track_content_top_margin = (HasLegend() ? 2.f : 1.f) * layout_->GetTrackContentTopMargin();
+
+  return layout_->GetTrackTabHeight() + track_content_top_margin + GetLegendHeight() +
+         GetGraphContentHeight() + layout_->GetTrackContentBottomMargin();
 }
 
 template <size_t Dimension>
 float GraphTrack<Dimension>::GetLegendHeight() const {
-  if (IsCollapsed() || Dimension <= 1) {
-    return 0;
-  }
   // This is a bit of an arbitrary choice - I don't want to introduce an additional layout
   // parameter just for the legend size, but it should be smaller than all regular textboxes
   // across Orbit.
-  return layout_->GetTextBoxHeight() / 2.f;
+  return HasLegend() ? layout_->GetTextBoxHeight() / 2.f : 0;
```

Comment:
I didn't see this before. If you want to introduce a new variable, consider in the future to add a method in Layout which simply returns GetTextBoxHeight()/2. In this way all the logic respect of sizes is in this class.

Reply:
I guess this comments might be added while Florian was improving this part.

I'm kind of agree with the comment here. The legend is only used in graph tracks, I prefer to not add and additional layout parameter.

Reply:
Sorry, I was not clear. If you DON'T want to introduce ... 

You can simply add a method without a parameter.